ui/src/app/pages/workspace/workspace-edit.spec.tsx

```diff
@@ -1100,138 +1015,123 @@ describe('WorkspaceEdit', () => {
       },
     });
 
-    const wrapper = component();
-    await waitOneTickAndUpdate(wrapper);
+    renderComponent();
 
-    const billingDropDown = wrapper
-      .find('[data-test-id="billing-dropdown"]')
-      .first();
+    expect(await screen.findByDisplayValue(freeTierRegex)).toBeInTheDocument();
 
-    expect(billingDropDown.props().value).toEqual('free-tier');
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.value)).toEqual([
-      'free-tier',
-      'user-billing',
-    ]);
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.label)).toEqual([
-      'Use All of Us initial credits - $34.56 left',
-      'User Billing',
-    ]);
-  });
-
-  it('should show free tier and user billing account when they grant billing scope when creating workspace', async () => {
-    mockHasBillingScope.mockImplementation(() => true);
-    workspaceEditMode = WorkspaceEditMode.Create;
-    const wrapper = component();
-    await waitOneTickAndUpdate(wrapper);
+    // Need to open the dropdown to view the billing options
+    await user.click(screen.getByTestId('billing-dropdown'));
 
-    const billingDropDown = wrapper
-      .find('[data-test-id="billing-dropdown"]')
-      .first();
+    expect(
+      screen.getAllByRole('option', {
+        name: freeTierRegex,
+        hidden: true,
+      })[0]
+    ).toBeInTheDocument();
 
-    expect(mockEnsureBillingScope).toHaveBeenCalledTimes(0);
-    expect(billingDropDown.props().value).toEqual('free-tier');
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.value)).toEqual([
-      'free-tier',
-      'user-billing',
-    ]);
-    // @ts-ignore
-    expect(billingDropDown.props().options.map((o) => o.label)).toEqual([
-      'Use All of Us initial credits - $33.33 left',
-      'User Billing',
-    ]);
+    expect(
+      screen.queryByRole('option', {
+        name: USER_BILLING_OPTION_REGEX,
+        hidden: true,
+      })
+    ).toBeInTheDocument();
   });
 
   it('should show User Provided Billing Account when user does not have permission on the billing account workspace is using', async () => {
     mockHasBillingScope.mockImplementation(() => true);
     workspaceEditMode = WorkspaceEditMode.Edit;
-    const wrapper = component();
-    await waitOneTickAndUpdate(wrapper);
-
-    const billingDropDown = wrapper
-      .find('[data-test-id="billing-dropdown"]')
-      .first();
+    renderComponent();
```

Comment:
I don't understand this test.  But this version seems to be a faithful copy.

Reply:
I think that this tests how a billing account is rendered if I don't have access to it. I am not entirely sure either though.

Reply:
ok, so the **current** billing account of the workspace would be an available option (no change) even if a user would otherwise not be able to select it - that makes sense